sinks.go

```diff
@@ -46,6 +46,20 @@ func (p *Pipe) CountLines() (int, error) {
 	return lines, p.Error()
 }
 
+// Slice return each lines of the pipe in a Slice or an error.
+// If there is an error reading the pipe, the pipe's error status
+// is also set.
+func (p *Pipe) Slice() ([]string, error) {
+	if p == nil || p.Error() != nil {
+		return nil, p.Error()
+	}
+	var slice []string
+	p.EachLine(func(line string, out *strings.Builder) {
+		slice = append(slice, line)
```

Comment:
This is probably a bit inefficient; how about getting the result of `p.String()` instead and splitting on newlines? Don't forget to make sure the last line also ends with a newline, if the contents of the pipe does.

Reply:
I am not sure of what to do when the pipe is empty.
Should we have a `Slice with 1 empty element` or an `empty Slice`?

And just to be sure, if the pipe ends with a new line, the last element of the Slice will be an empty String, right?

Reply:
Actually, when the pipe is empty, there are _no lines_, so I think it makes sense to return a slice with _no elements_, despite my previous comment.

If the pipe consists of a single newline, then the result of Slice() should be a slice with 1 element which is a string containing a newline, I think.